connection.go

```diff
@@ -144,6 +144,10 @@ func (mc *mysqlConn) interpolateParams(query string, args []driver.Value) (strin
 	buf = buf[:0]
 	argPos := 0
 
+	if len(args) != strings.Count(query, "?") {
+		return "", ErrArgsInvalid
```

Comment:
@methane 
I don't know when `\?` will appear,
It's look like `SELECT name FROM users WHERE name='\?'`?
i think that very uncommon

And i don't understand why `drvier.ErrSkip` safer